### PR TITLE
Explain serialized ASTs in ICFP response

### PR DESCRIPTION
**Pull request by brson**

### CODE REVIEW

**Comment by brson** [comment]
I just explained the _one big thing_ that serialized ASTs are for. The total metadata is used for lots of little reasons, but the reason for serializing ASTs specifically is to monomorphize.

File: icfp15-servo/response.txt [hunk 1]
```diff
@@ -8,7 +8,16 @@ Review 1
 ASTs within "compiled" binary? What's the point? When and how these ASTs
 are later used?
 
-TODO: ANSWER
+Rust serializes ASTs of items with type parameters into libraries so
+that when those items are later instantiated with specific types, the
+compiler can generate machine code that is optimized for those
+instances. The effect is as with C++ requiring templates to be written
+header files. We will clarify.
```
Comment (jdm) [inline icfp15-servo/response.txt:15]: written **in** header files.

**Comment by larsbergstrom** [comment]
Thanks much!

File: icfp15-servo/response.txt (hunk 1 above)
Comment (larsbergstrom) [inline icfp15-servo/response.txt:15]: lol, simultaneous comment+merge :-)

**Pull request merged**